RecoLocalTracker/SiPhase2VectorHitBuilder/plugins/SiPhase2RecHitMatcherESProducer.cc

```diff
@@ -0,0 +1,80 @@
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "RecoLocalTracker/Records/interface/TkPhase2OTCPERecord.h"
+
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderEDProducer.h"
+#include <memory>
+
+class SiPhase2RecHitMatcherESProducer : public edm::ESProducer {
+public:
+  SiPhase2RecHitMatcherESProducer(const edm::ParameterSet&);
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+  std::shared_ptr<VectorHitBuilderEDProducer> produce(const TkPhase2OTCPERecord&);
+
+private:
+  std::string name_;
+  std::shared_ptr<VectorHitBuilderEDProducer> matcher_;
+  edm::ParameterSet pset_;
+};
+
+SiPhase2RecHitMatcherESProducer::SiPhase2RecHitMatcherESProducer(const edm::ParameterSet& p) {
+  name_ = p.getParameter<std::string>("ComponentName");
+  pset_ = p;
+  setWhatProduced(this, name_);
+}
+
+std::shared_ptr<VectorHitBuilderEDProducer> SiPhase2RecHitMatcherESProducer::produce(
+    const TkPhase2OTCPERecord& iRecord) {
+  if (name_ == "SiPhase2VectorHitMatcher") {
+    matcher_ = std::make_shared<VectorHitBuilderEDProducer>(pset_);
+
+    edm::ESHandle<TrackerGeometry> tGeomHandle;
```

Comment:
Please use `ESGetToken` to get the input products in ESProducer. See examples in other ESProducers or
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ESProducer
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#With_token

Reply:
Done.